clcache.py

```diff
@@ -1600,57 +1574,89 @@ def processCompileRequest(cache, compiler, args):
 def processDirect(cache, objectFile, compiler, cmdLine, sourceFile):
     manifestHash = ManifestRepository.getManifestHash(compiler, cmdLine, sourceFile)
     manifestSection = cache.manifestRepository.section(manifestHash)
+    artifactSection = None
     with manifestSection.lock:
         manifest = manifestSection.getManifest(manifestHash)
-        if manifest is None:
-            return postprocessUnusableManifestMiss(
-                cache, objectFile, manifestSection, manifestHash, sourceFile, compiler, cmdLine,
-                Statistics.registerSourceChangedMiss)
+        if manifest:
+            for entryIndex, entry in enumerate(manifest.entries()):
+                # NOTE: command line options already included in hash for manifest name
+                try:
+                    includesContentHash = ManifestRepository.getIncludesContentHashForFiles(
```

Comment:
It would be interesting to move this computation out of the critical section as it usually shows in the profiling as one of the hotspots. It can be done by moving the if outside of the `with manifestSection.lock` but will complicate the code around touchEntry and setManifest.